### PR TITLE
Merge release/4.52 into trunk

### PR DESCRIPTION
**Pull request by mokagio**
Merging `release/4.52` into `trunk`.

Via intermediate branch `merge/release-4.52-into-trunk`, to help fix conflicts if any:
```
                            release/4.52  ----o-- - - -
                                               \
           merge/release-4.52-into-trunk        `---.
                                                     \
                                   trunk  ------------x- - -
```

---

@Automattic/apps-infrastructure this PR includes automation chances that I ported from WooCommerce iOS (see also https://github.com/woocommerce/woocommerce-ios/pull/13793) as part of the ongoing release automation modernization for Simplenote (see also internal ref paaHJt-6Zx-p2). I'd appreciate a look, even though the fact that everything worked is already good proof of the validity of the changes. Thanks!

If no one gets to it before the time to start the next code freeze, I'll admin merge it.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR has the <code>Releases</code> label: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: fastlane/Fastfile [hunk 1]
```diff
@@ -1028,8 +970,11 @@ def release_branch_name(release_version: release_version_current)
   "#{RELEASE_BRANCH_ROOT}#{release_version}"
 end
 
-def ensure_git_branch_is_release_branch
-  # Verify that the current branch is a release branch. Notice that `ensure_git_branch` expects a RegEx parameter
+def ensure_git_branch_is_release_branch!
+  # Verify that the current branch is a release branch.
+  #
+  # Notice that `ensure_git_branch` expects a RegEx parameter
```
Comment (mokagio) [inline fastlane/Fastfile]:
```suggestion
  # Verify that the current branch is a release branch.
  # Notice that `ensure_git_branch` expects a RegEx parameter.
```

File: fastlane/lanes/localization.rb [hunk 2]
```diff
@@ -178,10 +203,6 @@ def ensure_default_metadata_are_not_overridden(metadata_files_hash:)
     MSG
     UI.user_error!(override_not_allowed_message) if File.exist?(en_file_path)
   end
-
-  lane :lint_localizations do
-    ios_lint_localizations(input_dir: APP_RESOURCES_DIR, allow_retry: true)
-  end
```
Comment (mokagio) [inline fastlane/lanes/localization.rb:184]: Don't know how this ended up inside a method 😳

Comment (AliSoftware) [reply fastlane/lanes/localization.rb:184]: Ooof. Nice catch.

Probably bad indentation at some point in the past tricked us, and then we ran `rubocop -A` at some point later but didn't notice it despite being more obvious once indentation was fixed? 🤷

**Review by AliSoftware — APPROVED** [review]
Changes in `Fastfile` LGTM 👍

**Pull request merged**